### PR TITLE
Add specific errors for missing kind and version.

### PR DESCRIPTION
**Pull request by brendandburns**
Partially addresses #1908

### CODE REVIEW

**Comment by bgrant0607** [comment]
When would this happen?

IMO, we should be automatically setting kind and version by default since we have the information from the URL path.

**Comment by brendandburns** [comment]
It can happen client side with incomplete yaml files.

Filling in with defaults from paths is fine with me too.

Let's either merge this PR or close the open issue requesting this.

Brendan

File: pkg/conversion/error.go [hunk 1]
```diff
@@ -49,3 +49,35 @@ func IsNotRegisteredError(err error) bool {
...
+	_, ok := err.(*missingKindErr)
+	return ok
+}
+
+type missingVersionErr struct {
+	data string
+}
+
+func (k *missingVersionErr) Error() string {
+	return fmt.Sprintf("Object 'Version' is missing in '%s'", k.data)
```
Comment (bgrant0607) [inline pkg/conversion/error.go]: Should be apiVersion.

**Comment by bgrant0607** [comment]
Ok, this is independently useful.

File: pkg/conversion/error.go (hunk 1 above)
Comment (brendandburns) [inline pkg/conversion/error.go]: done.

**Comment by brendandburns** [comment]
Comment addressed.

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**